SU2_CFD/src/output/filewriter/CSurfaceFVMDataSorter.cpp

```diff
@@ -1115,11 +1129,36 @@ void CSurfaceFVMDataSorter::SortConnectivity(CConfig *config, CGeometry *geometr
   SU2_MPI::Allreduce(&nTotal_Surf_Elem, &nGlobal_Elem_Par, 1, MPI_UNSIGNED_LONG, MPI_SUM, MPI_COMM_WORLD);
 #endif
 
-  connectivity_sorted = true;
+  connectivitySorted = true;
 
 }
 
-void CSurfaceFVMDataSorter::SortSurfaceConnectivity(CConfig *config, CGeometry *geometry, unsigned short Elem_Type) {
+void CSurfaceFVMDataSorter::SortConnectivity(CConfig *config, CGeometry *geometry, const vector<string> &markerList) {
+
+  /*--- Sort connectivity for each type of element (excluding halos). Note
+   In these routines, we sort the connectivity into a linear partitioning
+   across all processors based on the global index of the grid nodes. ---*/
+
+  /*--- Sort volumetric grid connectivity. ---*/
+
+  SortSurfaceConnectivity(config, geometry, LINE         , markerList);
+  SortSurfaceConnectivity(config, geometry, TRIANGLE     , markerList);
+  SortSurfaceConnectivity(config, geometry, QUADRILATERAL, markerList);
+
+
+  unsigned long nTotal_Surf_Elem = nParallel_Line + nParallel_Tria + nParallel_Quad;
+#ifndef HAVE_MPI
+  nGlobal_Elem_Par   = nTotal_Surf_Elem;
+#else
+  SU2_MPI::Allreduce(&nTotal_Surf_Elem, &nGlobal_Elem_Par, 1, MPI_UNSIGNED_LONG, MPI_SUM, MPI_COMM_WORLD);
+#endif
```

Comment:
Well if I start with that now, I'll have to change it everywhere. Let's save that for a separate PR.

Reply:
Sure, it would be better to get rid of a lot of unnecessary preprocessor statements for MPI calls in a separate PR but that shouldn't be a free ticket introduce these things in new code. In COutput.cpp +559 you also use Allreduce without that "padding".

Reply:
+1 on this one, I've been replacing those things wherever I see them, better do something than do nothing.

Reply:
👍, yep, a lot of this is holdover from before we had the wrapper. Best practice is to hide the ifdefs within mpi_structure whenever possible!

Reply:
I agree with just doing them as you go